docs/graph-algorithms/prims-algorithm.mdx

```diff
@@ -0,0 +1,255 @@
+      // if the vertex is already in the set S, discard it otherwise
+      // choose another vertex nearest to selected vertex at step 1.
+
+      int min = INF;
+      int x = 0; // row number
+      int y = 0; // col number
+
+      for (int i = 0; i < V; i++) {
+        if (selected[i] == true) {
+          for (int j = 0; j < V; j++) {
+            // not in selected and there is an edge
+            if (!selected[j] && G[i][j] != 0) {
+              if (min > G[i][j]) {
+                min = G[i][j];
+                x = i;
+                y = j;
+              }
+            }
+          }
+        }
+      }
+      System.out.println(x + " - " + y + " :  " + G[x][y]);
+      selected[y] = true;
+      no_edge++;
+    }
+  }
+
+  public static void main(String[] args) {
+    PGraph g = new PGraph();
+
+    // number of vertices in grapj
+    int V = 5;
+
+    // create a 2d array of size 5x5
+    // for adjacency matrix to represent graph
+    int[][] G = { { 0, 9, 75, 0, 0 }, { 9, 0, 95, 19, 42 }, { 75, 95, 0, 51, 66 }, { 0, 19, 51, 0, 31 },
+        { 0, 42, 66, 31, 0 } };
+
+    g.Prim(G, V);
+  }
+}
+```
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Input :
+```
+
+Enter the elements into graph array:
+
+1 4 78 15                                                                                                                                                                            
+1 75 48 6                                                                                                                                                                            
+45 7 8 36                                                                                                                                                                            
+2 4 67 59   
+```
+
+#### Output :
+```
+
+Weight of ele, Edge of graph
+
+4  0 - 1                                                                                                                                                                             
+6  1 - 3                                                                                                                                                                             
+48  1 - 2 
+```
+
+#### Input :
+```
+
+Enter the elements into graph array :
+
+5  10  2  6
+4  17  19 3
+1  5   8  20
+15 1  16  7 
+```
+
+#### Output :
+```
+
+Weight of ele, Edge of graph
+
+2 0 -2
+5 2 -1
+3 1 -3
+```
+
+## Complexity Analysis
+- Time Complexity: `O(ElogV) where V no of vertices and E no of edges`
+- Space Complexity: `O(V) where V no of vertices`
+
+## Credits
+
+   - [Devara Khurma Venkata Kavyasree](https://github.com/kavyasree-123) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/ds/prims_algo.cpp)
+   - [Ricardo Prins](https://github.com/ricardoprins) for the [Java implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Java/graphs/Prim_Algorithm.java)
```

Comment:
It was worked on by Shubham. Here is his PR: https://github.com/TesseractCoding/NeoAlgo/pull/148

Reply:
@HarshCasper I have taken the code from the link which I have mentioned. And there are 4 to 5 prim's algorithms in Neoalgo